subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/DefaultConfigurationResolver.java

```diff
@@ -185,52 +186,67 @@ public ResolverResults resolveGraph(ResolveContext resolveContext) {
         DefaultResolvedArtifactsBuilder artifactsBuilder = new DefaultResolvedArtifactsBuilder(buildProjectDependencies, resolutionStrategy.getSortOrder());
         FileDependencyCollectingGraphVisitor fileDependencyVisitor = new FileDependencyCollectingGraphVisitor();
         ResolutionFailureCollector failureCollector = new ResolutionFailureCollector(componentSelectorConverter);
-        DependencyGraphVisitor graphVisitor = new CompositeDependencyGraphVisitor(newModelBuilder, localComponentsVisitor, failureCollector);
 
-        ImmutableList.Builder<DependencyArtifactsVisitor> visitors = new ImmutableList.Builder<>();
-        visitors.add(oldModelVisitor);
-        visitors.add(fileDependencyVisitor);
-        visitors.add(artifactsBuilder);
+        List<DependencyGraphVisitor> graphVisitors = new ArrayList<>();
+        graphVisitors.add(newModelBuilder);
+        graphVisitors.add(localComponentsVisitor);
+        graphVisitors.add(failureCollector);
```

Comment:
❓ Why replace this? I don't see any reason for it.

Reply:
In the previous version, we had a variable number of artifact visitors. Now, we always have 3 artifact visitors and there is a variable number of graph visitors.

Reply:
I don't understand how that changes what `List` implementation to use though. I don't have a preference for either implementation, just want to avoid unnecessary changes.

Looking at it again, I can see that these are different variables, so I suppose it would be mostly changing anyways. I just saw the change in implementation and wondered why it happened.

Reply:
I can change this back to an `ImmutableList.Builder` if that's your concern